### PR TITLE
Fix Mantis:15355: ECS: Use correct login prefix

### PR DESCRIPTION
**Pull request by Rillke**
- Do not look-up a MID by PID.
- The MID is not the same as the PID.
- The MID can't be reliably determined.  (c.f. comments for why)
- Instead, always use abreviation (abbr) provided by Auth request

### CODE REVIEW

**Comment by Rillke** [comment]
Will someone port this to trunc and 5_4 branches, or is that my obligation?

**Comment by mjansenDatabay** [comment]
Dear @Rillke ,

thanks a lot for your contribution. According to our [pull request guidelines](https://github.com/ILIAS-eLearning/ILIAS#pull-requests), an issue has to be filed [here](https://mantis.ilias.de/bug_report_page.php) when providing a bugfix as pull request.

Usually the component maintainer will cherry-pick a bugfix to the other branches.

**Comment by Rillke** [comment]
> an issue has to be filed here

@mjansenDatabay Thanks for pointing to the guidelines. Looking at #2056 and its Mantis entry, which describes an issue and not a bugfix, I assume that Mantis:15355 sufficient.

**Comment by mjansenDatabay** [comment]
@Rillke Sorry, I only checked the description and missed the Mantis ticket ID ;-)

**Comment by pawohl** [comment]
@mjansenDatabay Thanks for commenting. What prevents this pull request from being reviewed?

**Comment by mjansenDatabay** [comment]
@smeyer-ilias has to review this, he is the maintainer.

**Comment by Rillke** [comment]
Sooo ... @uni-halle I suggest you invest in having some people around in the communities of the open source products you make use of if you want to have fixed anything quickly. Maintaining custom patches is quite expensive, too.

**Comment by pawohl** [comment]
@mjansenDatabay I tried reaching out to @smeyer-ilias without success using Mantis and personal email asking what we can do (if it is money, we'd probably find a way for that, too, but takes some time).

**Comment by Rillke** [comment]
When you need a rebase, please let me know.

**Comment by kergomard** [comment]
@smeyer-ilias : Could you please give feedback, merge it or provide your objections?

**Comment by smeyer-ilias** [comment]
Sorry, for the late response. I will analyze this PR next week. 
Before I have to rebuild a new local ECS testing environment, since the current external environment cannot be used for the new stable release.

**Comment by Rillke** [comment]
That's great. If you need a test ECS account for ECS4 v1.2.2 Debian 9 installation, just let me know.

File: Services/WebServices/ECS/classes/class.ilAuthProviderECS.php [hunk 1]
```diff
@@ -201,33 +215,7 @@ public function validateHash()
...
-				{
-					$this->getLogger()->warning('Authentication failed with message: ' . $e->getMessage());
-					return false;
-				}
-			}
-			else
-			{
-				$this->abreviation = $auths->abbr;
-			}
+			$this->abreviation = $auths->abbr;
```
Comment (smeyer-ilias) [inline Services/WebServices/ECS/classes/class.ilAuthProviderECS.php:218]: Unfortunately $auths->abbr is marked as deprecated in https://www.freeit.de/documents/ecsa/ecsa.pdf chapter 3.7.3
So we cannot trust this value.

File: Services/WebServices/ECS/classes/class.ilAuthProviderECS.php [hunk 2]
```diff
@@ -201,33 +215,7 @@ public function validateHash()
 			
 			$this->getLogger()->dump($auths, ilLogLevel::DEBUG);
 
-			if($auths->pid)
-			{
-				try
-				{
-					include_once './Services/WebServices/ECS/classes/class.ilECSCommunityReader.php';
-					$reader = ilECSCommunityReader::getInstanceByServerId($this->getCurrentServer()->getServerId());
-					$part = $reader->getParticipantByMID($auths->pid);
```
Comment (smeyer-ilias) [inline Services/WebServices/ECS/classes/class.ilAuthProviderECS.php:210]: Yes, this was wrong. 
Using the $auths->pid as mid will procude wrong results if participants are assigned to more than one community.

Comment (smeyer-ilias) [reply Services/WebServices/ECS/classes/class.ilAuthProviderECS.php:210]: Instead I will add the new method $reader->getParticipantsByPid($pid) which will be used to receive the current abreviation.

**Review by smeyer-ilias — COMMENTED** [review]
Thanks again for analyzing this. 
As mentioned in the comments I close this PR. 

A new implementation is done in b6f52c180bfea502f97b96780a757d638ff482a2

**Pull request closed without merging**